test/serverExplorer.test.ts

```diff
@@ -0,0 +1,301 @@
+        expect(getStub).calledOnce;
+    });
+
+    test('serverExplorer.addServerOutput call should show ServerOutput', () => {
+        const getStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            show: () => {},
+            append: () => {}
+        });
+        serverExplorer.addServerOutput(ProcessOutput);
+        expect(getStub).calledOnce;
+    });
+
+    test('serverExplorer.refresh() should trigger serverExporer.getChildren() call for root node', () => {
+        const fireStub = sandbox.stub(EventEmitter.prototype, 'fire');
+        serverExplorer.refresh(serverHandle);
+        expect(fireStub).calledOnce;
+        expect(fireStub).calledOnceWith(serverHandle);
+    });
+});
+
+suite('serverExplorer.updateServer', () => {
+
+    let sandbox: sinon.SinonSandbox;
+    let serviceStub: sinon.SinonSandbox;
+    let serverStatusStub: sinon.SinonSandbox;
+    let getStub: sinon.SinonSandbox;
+    const clientStub: sinon.SinonStubbedInstance<RSPClient> = new RSPClient('somehost', 8080);
+    let serverExplorer: ServersViewTreeDataProvider;
+
+    const serverType: Protocol.ServerType = {
+        description: 'a type',
+        id: 'type',
+        visibleName: `the type`
+    };
+
+    const serverHandle: Protocol.ServerHandle = {
+        id: 'id',
+        type: serverType
+    };
+
+    const serverStop = {
+        collapsibleState: 0,
+        label: `id:the type(Stopped)`,
+        contextValue: 'Stopped'
+    };
+
+    const serverStart = {
+        collapsibleState: 0,
+        label: 'id:the type(Started)',
+        contextValue: 'Started'
+    };
+
+    const serverStopping = {
+        collapsibleState: 0,
+        label: 'id:the type(Stopping)',
+        contextValue: 'Stopping'
+    };
+
+    const serverStarting = {
+        collapsibleState: 0,
+        label: 'id:the type(Starting)',
+        contextValue: 'Starting'
+    };
+
+    const serverUnknown = {
+        collapsibleState: 0,
+        label: 'id:the type(Unknown)',
+        contextValue: 'Unknown'
+    };
+
+    const stateChange: Protocol.ServerStateChange = {
+        server: serverHandle,
+        state: 1
+    };
+
+    setup(() => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        sandbox = sinon.createSandbox();
+        serviceStub = sandbox.stub(serverExplorer.servers, 'get').returns({
+            stateChange
+        });
+        serverStatusStub = sandbox.stub(serverExplorer.serverStatus, 'set').returns({
+            stateChange
+        });
+        getStub = sandbox.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {}
+        });
+        sandbox.stub(serverExplorer.servers, 'values').returns(Array.from([serverHandle]));
+        sandbox.stub(serverExplorer, 'refresh');
+    });
+
+    teardown(() => {
+        sandbox.restore();
+    });
+
+    test('call should update server state to received in state change event (Stopped)', () => {
```

Comment:
I don't see where you are actually checking that state is changed from original one to one from state change event. That seems to be the problem for all the tests in this suite.